src/Ser/SerCsv.jl

```diff
@@ -17,6 +26,144 @@ function wrap_value(s::AbstractString)
+            has_simple = true
+        elseif iscomposite(type)
+            has_composite = true
+        end
+    end
+    return has_simple ⊻ has_composite
+end
+
+function could_flatten(::Type{T})::Bool where {T}
+    return if issimple(T)
+        false
+    elseif isnull(T)
+        false
+    elseif iscomposite(T)
+        true
+    elseif isunion(T)
+        is_valid_union(T) && !isnothing(extract_composite(T))
+    else
+        false
+    end
+end
+
+function extract_composite(::Type{T}) where {T}
+    return if iscomposite(T)
+        T
+    elseif isunion(T)
+        types = Base.uniontypes(T)
+        index = findfirst(iscomposite, types)
+        isnothing(index) ? nothing : types[index]
+    else
+        throw(CsvSerializationError("Parameter of type $T must be a Union type or Composite Type."))
+    end
+end
+
+function null_count(::Type{T}) where {T}
+    count = 0
+    for type in fieldtypes(T)
+        count += could_flatten(type) ?  null_count(extract_composite(type)) : 1
+    end
+    return count
+end
+
+function row_values(buf::IOBuffer, data::T; delimiter::String = ",") where {T}
+    count = 1
+    for (name, type) in zip(fieldnames(T), fieldtypes(T))
+        if isunion(type) && !is_valid_union(type)
+            throw(CsvSerializationError("Unsupported Union Type in field $name::$type of type $T."))
+        end
+        if count != 1
+            print(buf, delimiter)
+        end
+        value = getproperty(data, name)
+        if could_flatten(type)
+            if isnothing(value) || ismissing(value)
+                print(buf, repeat(delimiter, null_count(extract_composite(type)) - 1))
+            else
+                row_values(buf, value)
+            end
+        else
+            print(buf, value_to_string(value))
+        end
+        count += 1
+    end
+end
+
+function csv_headers(::Type{T})::Vector{String} where {T}
+    headers = Vector{String}()
+    for (name, type) in zip(fieldnames(T), fieldtypes(T))
+        if isunion(type) && !is_valid_union(type)
+            throw(CsvSerializationError("Unsupported Union Type in field $name::$type of type $T."))
+        end
+        if could_flatten(type)
+            for header in csv_headers(extract_composite(type))
+                push!(headers, String(name) * "_" * header)
+            end
+        else
+            push!(headers, String(name))
+        end
+    end
+    return headers
+end
+
+function to_csv(
+    data::Vector{T};
+    delimiter::String = ",",
+    headers::Vector{String} = String[],
+    with_names::Bool = true,
+)::String where {T}
+    buf = IOBuffer()
+    
+    if with_names
+        t_cols = if isempty(headers)
+            csv_headers(T)
+        elseif length(headers) == null_count(T)
+            headers
+        else
+            throw(DimensionMismatch("The dimensions of custom headers do not match the dimensions of the output."))
+        end
+
+        join(buf, t_cols, delimiter)
+        println(buf)
```

Comment:
Isn't it a debug statement?

Reply:
Of course not, `buf` is a `IOBuffer`, `println()` just append a `\n` to `buf` here